### PR TITLE
fix: [CDS-42834]: manifest collection helm v2 changes

### PR DESCRIPTION
**Pull request by achyuth-harness**
## Describe your changes

We have made the fix in above pr to not delete anything if the cache directory is empty, which is the case for helm v2. We have also ensured that we don't add repo flags for helm v2 as it is not supported

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37213)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Vulnerability 1 Vulnerabilities 
 
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37213)

File: 910-delegate-service-driver/src/main/java/io/harness/service/impl/DelegateSyncServiceImpl.java [hunk 1]
```diff
@@ -91,8 +91,10 @@ public <T extends ResponseData> T waitForTask(
           syncTaskWaitMap.computeIfAbsent(taskId, k -> new AtomicLong(currentTimeMillis() + timeout.toMillis()));
       synchronized (endAt) {
         do {
+          log.info("Waiting for sync task {} to complete", taskId);
```
Comment (gaurav-jain1) [inline 910-delegate-service-driver/src/main/java/io/harness/service/impl/DelegateSyncServiceImpl.java:94]: log.info's can be too much here. let's remove both of these.

**Review by achyuth-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37213)* status: 6 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`910-delegate-service-driver/src/main/java/io/harness/service/impl/DelegateSyncServiceImpl.java` line 94 at r2](https://reviewable.io/reviews/harness/harness-core/37213#-NBI7JOsSUe41j89PFC6-r2-94:-NBIUpIHBjYm64hRckjW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2d4355a4b9d4f3c22250756d624d8f681e5e82f2/910-delegate-service-driver/src/main/java/io/harness/service/impl/DelegateSyncServiceImpl.java#L94)):*
<details><summary><i>Previously, gaurav-jain1 (Gaurav Jain) wrote…</i></summary><blockquote>

log.info's can be too much here. let's remove both of these.
</blockquote></details>

Done.

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 4 of 4 files at r1, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37213)* status: :shipit: complete! ✓

**Pull request merged**